lib/cms/hippo_importer/base.rb

```diff
@@ -0,0 +1,86 @@
+module Cms
+  module HippoImporter
+    class Base
+      attr_reader :records
+
+      def initialize(data:, docs: [], to: 'en', parser: HippoXmlParser)
+        @records = parser.parse(data, [hippo_type])
+        @_site = to
+        @docs = docs
+      end
+
+      def hippo_type
```

Comment:
Yes, agreed in parts. Following this idea, I will extract the line from initialize to a method called records and remove the attr_reader records. But the hippo_type is needed, because the subclass will overwrite, because there are different types on hippo corresponding to a different layouts on our database. Like the CorporatePage class for example.